ipaserver/install/dns.py

```diff
@@ -143,6 +145,16 @@ def install_check(standalone, api, replica, options, hostname):
                 if hst not in e.kwargs['ns']:
                     raise ValueError(str(e))
 
+    # https://pagure.io/freeipa/issue/8700
+    # https://www.freedesktop.org/software/systemd/man/systemd-resolved.service.html
+    # Temporary copy the resolv.conf file containing the upstream DNS servers
+    # into /etc/resolv.conf so that systemd-resolved does not interfere with
+    # reverse zone detection.
+    if os.path.islink(paths.RESOLV_CONF):
+        if os.readlink(paths.RESOLV_CONF) == paths.RESOLV_CONF_STUB_RESOLVED:
+            os.unlink(paths.RESOLV_CONF)
+            shutil.copyfile(paths.RESOLV_CONF_RESOLVED, paths.RESOLV_CONF)
```

Comment:
Would it be possible to update the link to `paths.RESOLV_CONF_RESOLVED` instead of removing it and copying the file?

Reply:
That was my first idea and is the 3rd way to configure systemd-resolved.

However, systemd-resolved then merges the upstream DNS list with 127.0.0.1 (the FreeIPA DNS provided via a NetworkManager configuration snippet).
The resulting resolv.conf contains something like:
```
nameserver 127.0.0.1
nameserver UPSTREAM_DNS_1
nameserver UPSTREAM_DNS_2
domain FOO
```
which is somewhat brittle and misconfigured.
This is why I decided on the fourth way e.g. provide a resolv.conf that works for us during install.